### PR TITLE
Issue #2502433 by Pol, Fabianx: Update Drupal 8 Core library and Symfony Component

### PR DESCRIPTION
**Pull request by drupol**
Ticket: https://www.drupal.org/node/2502433

Update Drupal 8 Core library and Symfony Component

### CODE REVIEW

File: lib/Drupal/Core/Database/Connection.php [hunk 1]
```diff
@@ -2,233 +2,1302 @@
 
 /**
  * @file
- * Contains \Drupal\Core\Database\Connection.
+ * Definition of Drupal\Core\Database\Connection
```
Comment (dawehner) [inline lib/Drupal/Core/Database/Connection.php]: This entire file change is kinda wrong

Comment (drupol) [inline lib/Drupal/Core/Database/Connection.php]: I will check this out tomorrow. I didn't know you edited some files.

File: lib/Symfony/Component/HttpFoundation/RequestStack.php [hunk 2]
```diff
@@ -0,0 +1,103 @@
+<?php
```
Comment (fabianx-ai) [inline lib/Symfony/Component/HttpFoundation/RequestStack.php]: The RequestStack sneaked in again ;).

File: lib/Drupal/Core/StringTranslation/TranslationWrapper.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
+<?php
```
Comment (fabianx-ai) [inline lib/Drupal/Core/StringTranslation/TranslationWrapper.php]: Not used

File: lib/Drupal/Core/StringTranslation/StringTranslationTrait.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
+<?php
```
Comment (fabianx-ai) [inline lib/Drupal/Core/StringTranslation/StringTranslationTrait.php]: Not used.

File: lib/Drupal/Core/StringTranslation/TranslationInterface.php [hunk 5]
```diff
@@ -70,19 +70,66 @@ public function translate($string, array $args = array(), array $options = array
...
+    * Returns the number of plurals supported by a given language.
+    *
+    * @param null|string $langcode
+    *   (optional) The language code. If not provided, the current language
+    *   will be used.
+    *
+    * @return int
+    *   Number of plural variants supported by the given language.
+    */
+   public function getNumberOfPlurals($langcode = NULL);
```
Comment (fabianx-ai) [inline lib/Drupal/Core/StringTranslation/TranslationInterface.php:133]: https://github.com/LionsAd/service_container/blob/7.x-1.x/src/StringTranslation/StringTranslation.php

Will need updating for the new methods. Thanks Scrutinizer!

Comment (fabianx-ai) [inline lib/Drupal/Core/StringTranslation/TranslationInterface.php:133]: It also means we miss test coverage for 'string_translation' service :(.

File: lib/Drupal/Core/StringTranslation/TranslationWrapper.php (hunk 3 above)
Comment (drupol) [inline lib/Drupal/Core/StringTranslation/TranslationWrapper.php]: Used in RfcLogLovel. Sure about removing it ?

File: lib/Drupal/Core/StringTranslation/TranslationInterface.php (hunk 5 above)
Comment (drupol) [inline lib/Drupal/Core/StringTranslation/TranslationInterface.php:133]: If we commit https://github.com/LionsAd/service_container/pull/28 we'll be able to test it properly.

**Comment by fabianx-ai** [comment]
@drupol Lets add a follow-up issue on drupal.org to add a patched StringTranslationWrapper (with the trait included statically) to support RfcLogLevels correctly. (and add a test for rfc log levels)

File: tests/src/StringTranslation/StringTranslationTest.php [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+ * @file
+ * Contains \Drupal\Tests\service_container\StringTranslation\StringTranslationTest
+ */
+
+namespace Drupal\Tests\service_container\StringTranslation;
+
+use Drupal\service_container\StringTranslation\StringTranslation;
+
+/**
+ * @coversDefaultClass \Drupal\service_container\StringTranslation\StringTranslationTest
```
Comment (fabianx-ai) [inline tests/src/StringTranslation/StringTranslationTest.php]: Should be:

\Drupal\service_container\StringTranslation\StringTranslation

Then tests pass.

Comment (fabianx-ai) [inline tests/src/StringTranslation/StringTranslationTest.php]: @drupol ^^ This is the correct fix.

**Comment by fabianx-ai** [comment]
@drupol Easiest, run:

./tests/run-tests.sh

locally before pushing :)

**Comment by drupol** [comment]
True, I still do not have the reflex to do it automatically.

File: lib/Drupal/Core/StringTranslation/TranslationInterface.php (hunk 5 above)
Comment (fabianx-ai) [inline lib/Drupal/Core/StringTranslation/TranslationInterface.php:133]: Opened https://www.drupal.org/node/2502525

**Pull request merged**